pkg/kp/kv.go

```diff
@@ -50,6 +52,32 @@ func (err KVError) Error() string {
 	return fmt.Sprintf("%s failed for path %s", err.Op, err.Key)
 }
 
+func (c consulStore) Put(key, value string) (time.Duration, error) {
+	value = appendTimeStamp(value)
+	keyPair := &api.KVPair{
+		Key:   key,
+		Value: []byte(value),
+	}
+
+	writeMeta, err := c.client.KV().Put(keyPair, nil)
+	var retDur time.Duration
+	if writeMeta != nil {
+		retDur = writeMeta.RequestTime
+	}
+	if err != nil {
+		return retDur, err
+	}
+	return retDur, nil
+}
+
+func (c consulStore) Get(key string) (*api.KVPair, error) {
```

Comment:
We shouldn't expose primitives of the Consul API SDK in the `kp` package. This is intended to be a P2-specific key value wrapper that is abstracted from the underlying storage mechanism.